commit ca9015f4dead2c97b48f5a6933631b0a448b65b9
Author: Jia Tan <jiat0218@gmail.com>
Date:   Thu Feb 1 16:06:29 2024 +0800

    liblzma: Check HAVE_USABLE_CLMUL before omitting CRC64 table.
    
    If liblzma is configured with --disable-clmul-crc
    CFLAGS="-msse4.1 -mpclmul", then it will fail to compile because the
    generic version must be used but the CRC tables were not included.

diff --git a/src/liblzma/check/crc64_table.c b/src/liblzma/check/crc64_table.c
index 0ae9dda8..806a9ce3 100644
--- a/src/liblzma/check/crc64_table.c
+++ b/src/liblzma/check/crc64_table.c
@@ -1,38 +1,38 @@
 ///////////////////////////////////////////////////////////////////////////////
 //
 /// \file       crc64_table.c
 /// \brief      Precalculated CRC64 table with correct endianness
 //
 //  Author:     Lasse Collin
 //
 //  This file has been put into the public domain.
 //  You can do whatever you want with this file.
 //
 ///////////////////////////////////////////////////////////////////////////////
 
 #include "common.h"
 
 
 // FIXME: Compared to crc_common.h this has to check for __x86_64__ too
 // so that in 32-bit builds crc64_x86.S won't break due to a missing table.
-#if (defined(__x86_64__) && defined(__SSSE3__) \
+#if defined(HAVE_USABLE_CLMUL) && ((defined(__x86_64__) && defined(__SSSE3__) \
 			&& defined(__SSE4_1__) && defined(__PCLMUL__)) \
-		|| (defined(__e2k__) && __iset__ >= 6)
+		|| (defined(__e2k__) && __iset__ >= 6))
 #	define X86_CLMUL_NO_TABLE 1
 #endif
 
 
 #ifdef X86_CLMUL_NO_TABLE
 // No table needed. Use a typedef to avoid an empty translation unit.
 typedef void lzma_crc64_dummy;
 
 #else
 // Having the declaration here silences clang -Wmissing-variable-declarations.
 extern const uint64_t lzma_crc64_table[4][256];
 
 #	if defined(WORDS_BIGENDIAN)
 #		include "crc64_table_be.h"
 #	else
 #		include "crc64_table_le.h"
 #	endif
 #endif